lib/qiita/markdown/filters/user_input_sanitizer.rb

```diff
@@ -2,30 +2,90 @@ module Qiita
   module Markdown
     module Filters
       # Sanitizes raw markdown input if :strict context is given.
-      class UserInputSanitizer < HTML::Pipeline::TextFilter
+      class UserInputSanitizer < HTML::Pipeline::Filter
+        class AttributeTransformer
+          FILTERS = {
+            "a" => {
+              "class" => %w[autolink],
+              "rel" => %w[footnote url],
+              "rev" => %w[footnote],
+            },
+            "sup" => {
+              "id" => /^fnref\d+$/,
+            },
+            "li" => {
+              "id" => /^fn\d+$/,
+            },
+          }.freeze
+
+          def self.call(*args)
+            new(*args).transform
+          end
+
+          def initialize(env)
+            @env = env
+          end
+
+          def transform
+            return unless FILTERS.key?(name)
+            FILTERS[name].each_pair do |attr, pattern|
+              transform_attribute(attr, pattern) if node.attributes.key?(attr)
+            end
+          end
+
+          private
+
+          def transform_attribute(attr, pattern)
+            node.attributes[attr].value = node.attributes[attr].value.split.select do |value|
```

Comment:
You can write as:

* `node.attributes[attr].value` -> `node[attr]`
* `node.attributes[attr].value =` -> `node[attr] =`

http://www.rubydoc.info/gems/nokogiri/1.8.0/Nokogiri/XML/Node#[]-instance_method